### PR TITLE
[Docs] links to `logits-processor-zoo`

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

Adds links to [`logits-processor-zoo`](https://github.com/NVIDIA/logits-processor-zoo) on relevant parts of our `generate` docs. 

`logits-processor-zoo` is an external library containing custom `LogitsProcessors`, and part of a recent [HF blog post](https://huggingface.co/blog/logits-processor-zoo).

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
Oh did not know about this lib! Nice 🚀
